src/lib.rs

```diff
@@ -13,6 +13,8 @@
 #![feature(main)]
 #![feature(const_in_array_repeat_expressions)]
 #![feature(ptr_wrapping_offset_from)]
+#![feature(try_reserve)]
+#![feature(alloc_layout_extra)]
```

Comment:
Don't we want to not add nightly features?

If we do add them, be sure to update https://github.com/immunant/rexpat/issues/18

Reply:
Reading https://github.com/rust-lang/rust/issues/55724#issuecomment-552135444 makes me think they'll stabilize more of `Layout` soon. There's also been some recent discussion on [stabilizing `try_reserve`](https://github.com/rust-lang/rust/issues/48043).

Reply:
One alternative Stephen and I were discussing earlier is to have a `nightly` or `fallible` feature that conditionally enables fallible allocation, and fall back to panics if building for nightly Rust. We really, really need the nightly features for `runtests`, but could allow panics outside of that.

Reply:
I just realized something else: `alloc_layout_extra` is temporary, since once we replace all pointers with slices/boxes/etc we won't need to manually allocate any memory, so that feature will go away. We'll still need `try_reserve`, but we can enable that conditionally.